### DISCUSSION TITLE
"Failed to compile fragment shader." error on Chrome Android 6.0.1

### DISCUSSION THREAD

**Discussion by samplezzz** [vladmandic/human#271] [Q&A] [answered]
Hi Vlad,

Running on mobile Chrome on Android 6.0.1, the library is throwing the following error (line 23181 in human lib ver 2.7.3):
`  if (e.getShaderParameter(n, e.COMPILE_STATUS) === false)
    throw vv(t, e.getShaderInfoLog(n)), new Error("Failed to compile fragment shader.");`

Is this something you are aware of when using the lib on such old devices? Could this be fixed?

Greets,
Matt

**Comment by vladmandic** [comment, marked as answer]
Typically that means you're running out of GPU memory. WebGL uses much more memory than CPU models. And if it's an old device, it's unlikely you have good GPU anyhow, so you might as well try WASM backend instead.

Otherwise, try reducing which modules are enabled.

**Comment by samplezzz** [comment]
I tried moving to WASM with the following config (using latest Human 2.8.1):
` backend: 'wasm',
  wasmPath: 'https://cdn.jsdelivr.net/npm/@tensorflow/tfjs-backend-wasm@3.18.0/dist/'`

but getting the following error at `human.esm.js:2857:17`:

Uncaught (in promise) RuntimeError: Aborted(LinkError: WebAssembly.instantiate(): Import #4 module="env" function="emscripten_get_heap_max" error: function import requires a callable). Build with -s ASSERTIONS=1 for more info.

**Comment by vladmandic** [comment]
ah, thats 99% a binary incopatibility - you may need to rebuild wasm binaries for your cpu type, but thats a pain for a mobile device. sorry :(

**Reply by Eduard-Serra-Fourthline** [reply]
Hi Vlad,
I have noticed that the size of the facemesh model got almost halved.
Will it still be able to run on as many devices as before?
I am using WebGL backend.

**Reply by vladmandic** [reply]
Facemesh was quantized as f16, that brought size to half compared to f32. But it still uses same amount of memory as javascript does not differentiate between different float types internally.
